### ISSUE TITLE
New config: Problems with setting c.bindings.commands

### ISSUE THREAD

**Issue by noctuid** [qutebrowser/qutebrowser#2976]
The examples `c.aliases['foo'] = ':message-info foo'` and `c.hints.next_regexes.append(r'\bvor\b')`  work for me. On the other hand, if I try something like `c.bindings.commands['normal'] = ...` or `c.bindings.commands['normal'].update(...)`, there is no effect. `c.bindings.commands = ...` works fine though. 

As for why I want to do this instead of using `config.bind`, I want to swap `;` and `:` without manually rebinding `;<keys>`, so I'm extracting all the entries that start with `;` and changing them to start with `:` instead (is there a better way to swap keys?).

I'm using the latest version (v0.11.0-808-g046a16f92).

**Comment by jgkamat** [comment]
I took a quick look, and I don't think there's an API to read bindings currently.

1. `c.bindings.defaults = {}` currently errors out for me
2. `print(c.bindings.defaults)` errors out for me  

In terms of implementation, currently `config` is of type `ConfigAPI`, which stores key config seperate from 'normal config'. [However, it dosen't seem to currently provide a way to read from key                       configuration](https://github.com/qutebrowser/qutebrowser/blob/master/qutebrowser/config/configfiles.py#L123-L170). However, the underlying `KeyConfig` object seems to [have functions for getting bindings for a        mode](https://github.com/qutebrowser/qutebrowser/blob/master/qutebrowser/config/config.py#L143-L151), so this is probably possible to add fairly easily.

A **horrible way to do this which you shouldn't use because it's not documented** is here:
```
normal_key_map = config._keyconfig.get_bindings_for("normal")
entries = list(map(lambda x: (x, normal_key_map[x]),
                   filter(lambda x: x.startswith(":") or x.startswith(";"),
                          normal_key_map.keys())))
for entry in entries:
    config.unbind(entry[0], mode="normal")
for entry in entries:
    if entry[0][0] == ":":
        entry = (entry[0].replace(":", ";"), entry[1])
    elif entry[0][0] == ";":
        entry = (entry[0].replace(";", ":"), entry[1])
    config.bind(entry[0], entry[1], mode="normal")
```

Interestingly, `unbind` seems to be having no effect for me when `config.load_autoconfig = True` (default). When that's false, it seems to work fine. Hopefully that will be fixed if #2975 is addressed.

**Comment by noctuid** [comment]
It should be `c.bindings.default`. And this works fine for me:

```
hints_dict = dict()
for k, v in c.bindings.default['normal'].items():
    if k.startswith(';'):
        hints_dict[re.sub(r'^;(.*)', r':\1', k)] = v
hints_dict = {'normal': {**c.bindings.commands['normal'], **hints_dict}}
c.bindings.commands = hints_dict
```

**Comment by The-Compiler** [comment]
The easiest way to do this is `c.bindings.key_mappings[':'] = ';'` :wink: Though I'm not sure how that'll act outside normal mode.

@jgkamat The keyconfig object is just an API over keys, but they are stored in `bindings.commands` in the "normal" config. However, it should be `c.bindings.default`, not `.defaults` - I fixed the docs in d4709f7c44b12fc71bc0754ee81a83114f59f94b.

As for the original issue: It's possible that the code doesn't handle mutating of the nested dicts properly - I'll have to take a look.

**Comment by The-Compiler** [comment]
This seems to works fine - I'm going to guess it was fixed by #2999.
